File: 백준/Silver/2485. 가로수/가로수.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int euclidean(int a, int b);
int main()
{
int n;
cin >> n;
vector<int> tree;
for (int i = 0; i < n; i++) {
int u;
cin >> u;
tree.push_back(u);
}
sort(tree.begin(), tree.end());
vector<int> distance;
for (int i = 0; i < n - 1; i++) {
distance.push_back(tree[i + 1] - tree[i]);
}
int gcd = distance[0];
for (int i = 1; i < n - 1; i++) {
gcd = euclidean(gcd, distance[i]);
}
int cnt = 0;
for (int i = 0; i < n - 1; i++) {
cnt += (distance[i] / gcd) - 1;
}
cout << cnt;
}
int euclidean(int a, int b) {
if (b == 0) {
return a;
}
return euclidean(b, a % b);
}

File: 백준/Silver/2485. 가로수/가로수.java
import java.util.*;
import java.io.*;

class Main{
    private static int gcd(int a, int b){
        if(b==0) return a;
        else return gcd(b,a%b);
    }
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] tree = new int[n];
        for(int i = 0; i<n; i++){
            tree[i] = Integer.parseInt(br.readLine());
        }
        
        int[] distance = new int[n-1];
        for(int i = 0; i<n-1; i++){
            distance[i] = tree[i+1] - tree[i];
        }
        
        int g = distance[0];
        for(int i = 1; i<n-1; i++){
            g = gcd(g, distance[i]);
        }
        
        int cnt = 0;
        for(int i = 0; i<n-1; i++){
            cnt += distance[i]/g -1;
        }
        
        System.out.println(cnt);
        
    }
}